content/blogposts/2022/release-3.40.md

```diff
@@ -0,0 +1,49 @@
+    url: https://github.com/sourcegraph/sourcegraph/pull/34595
+    category: Search
+  - description: '`-language` is a valid filter, but the web app previously displayed it as invalid. This bug has been fixed to reflect it as valid.'
+    url: https://github.com/sourcegraph/sourcegraph/pull/34949
+    category: Search
+  - description: '`search.largeFiles` now supports recursive globs. For example, it is now possible to specify a pattern like `**/*.lock` to match a lock file anywhere in a repository.'
+    url: https://docs.sourcegraph.com/code_search/explanations/search_details
+    category: Search
+  - description: 'The `setRepositoryPermissionsUnrestricted` mutation was added, which allows you to explicitly mark a repo as available to all Sourcegraph users.'
+    url: https://github.com/sourcegraph/sourcegraph/pull/35378
+    category: Admin
+---
+
+Sourcegraph 3.40 is now available! Here are some highlights from this release:
+
+## Search your Python dependencies
+
+Sourcegraph [dependencies search](https://docs.sourcegraph.com/code_search/how-to/dependencies_search) now supports Python packages. You can now search through the Python dependencies of your repositories to easily identify the root cause of an error coming from the source code of a dependency or determine if any of your repository dependencies has a security vulnerability. We currently have support for poetry.lock and pipfile.lock files. Check out [this example search on Sourcegraph Cloud](https://sourcegraph.com/search?q=context:global+repo:deps%28%5Egithub%5C.com/textualize/rich%24%29+&patternType=literal).
+
+## Local navigation for the top 8 languages is now faster and more accurate
+
+We [previously improved navigation speed for Java code](https://about.sourcegraph.com/blog/release/3.39/#Local-navigation-for-Java-code-is-now-faster-and-more-accurate) in Sourcegraph 3.39. We’ve now made the same improvements for Go, C#, Python, JavaScript, TypeScript, C/C++, and Ruby, so you can navigate your code more quickly and precisely.
+
+## Batch Changes on Bitbucket Cloud
+
+Batch Changes is now available on Bitbucket Cloud. Batch Changes was previously only available on self-hosted deployments of Bitbucket; now, you can programmatically define changes across all of your Bitbucket repositories.
+
+## Share links to code insights
+
+[img placeholder]
+
+You can now link to individual code insight charts directly via the context menu on each insight. When you click to share an insight, the product indicates what other groups – global or individual Sourcegraph organizations – can see the insight. To share an insight chart, it will need to be on at least one global or org-wide dashboard, and the share link will enforce the permissions of the dashboard(s) with the insight. You can read more about [Code Insights viewing & sharing in our docs](https://docs.sourcegraph.com/code_insights/explanations/viewing_code_insights).
```

Comment:
@iskyOS @Joelkw - can you add a sentence on why this is valuable/why someone would want to share an insight chart and who they would share it with?

Reply:
```suggestion
You can now link to individual code insight charts directly via the context menu on each insight, to share individual insights with teammates rather than just whole dashboard pages. When you click to share an insight, it indicates which other groups of users can see the insight. To share an insight chart, it will need to be on at least one global or org-wide dashboard, and the share link will enforce the permissions of the dashboard(s) with the insight. You can read more about [Code Insights link sharing](https://docs.sourcegraph.com/code_insights/explanations/viewing_code_insights#sharing-links-to-individual-insights).
```

Reply:
@abschu does that cover what you're asking for? The TL;DR is that this is literally "share insights" so there are a long tail of uses – share an insight with a manager, with a teammate, to debug the insight, to link people to the individual insights, etc. Feel free to wordsmith it differently if you prefer!

Reply:
```suggestion
You can now link to individual Code Insights charts directly via the context menu on each insight, to share individual insights with teammates rather than entire dashboards. When you click to share an insight, it will indicate which other groups of users can see the insight. To share an insight chart, it will need to be on at least one global or org-wide dashboard, and the share link will enforce the permissions of the dashboard(s) with the insight. You can read more about [Code Insights link sharing](https://docs.sourcegraph.com/code_insights/explanations/viewing_code_insights#sharing-links-to-individual-insights).
```